pkg/discovery/processor/cluster_processor.go

```diff
@@ -65,68 +71,107 @@ func (processor *ClusterProcessor) ConnectCluster() error {
 	return nil
 }
 
+// Perform a single node validation
+func (processor *ClusterProcessor) checkNodesWorker(work chan *v1.Node, done chan bool, index int) {
+	glog.Infof("node verifier worker %d starting", index)
```

Comment:
While not required, it might pay to collect and log a few simple summary stats, e.g., #nodes in cluster, #nodes failed while waiting for one to succeed.